src/dmd/root/filename.d

```diff
@@ -883,11 +882,12 @@ nothrow:
         }
 
         version (Windows)
-            const r = _mkdir(path.toDString);
+            const r = _mkdir(path);
         version (Posix)
         {
             errno = 0;
-            const r = mkdir(path, (7 << 6) | (7 << 3) | 7);
+            assert(path.ptr[path.length] == 0); //must be 0 terminated
```

Comment:
Asserting on 0 is not wholly reliable, because that memory might be changed by someone else between when the check is made and when it is used.

Reply:
Pushing a safe version of this that incurs an extra allocation/deallocation, but that's probably small relative to the cost of mkdir itself.